### ISSUE TITLE
Jenkins should create a DC/RC/Job to manage the slave pod lifecycle

### ISSUE THREAD

**Issue by jfchevrette** [openshiftio/openshift.io#2351]
At the moment jenkins creates pod objects to run the slave. This puts the burden of managing the pods lifecycle on the jenkins plugin rather than on the Openshift side.

There was a networking issue with the cluster earlier today and jenkins kept trying to create jenkins slaves even though previous attempts failed. Some namespaces had over 10 jenkins slaves in a bad state at one time. Jenkins still kept trying to create new ones hoping for one to eventually come up.

We should let openshift manage the pod lifecycle. This will ensure that only one pod runs at any time and Openshift will attempt to restart or retry if the pod fails for some reason. There are many parameters to tweak that at the DC/RC/Job objects level.

**Comment by kishansagathiya** [comment]
> At the moment jenkins creates pod objects to run the slave.

@hrishin Is this something done by sync plugin?

**Comment by hrishin** [comment]
@kishansagathiya yes its Jenkins kubernetes plugin (not sync plugin), responsible for spinning up slaves.

**Comment by kishansagathiya** [comment]
@hrishin do you mean https://github.com/jenkinsci/kubernetes-pipeline-plugin?

**Comment by hrishin** [comment]
@kishansagathiya here you go https://github.com/jenkinsci/kubernetes-plugin. We are using this upstream one.

**Comment by kishansagathiya** [comment]
@jfchevrette @hrishin 
I see two issues here 
- [ ] Because of the networking issue Jenkins thinks that no slave pod is up
- [ ] Kubernetes plugin does not clean up pods

**Comment by chmouel** [comment]
new osio pipeline : 

 https://github.com/fabric8io/osio-pipeline.git

and the nodejs support in current pipeline : 

https://github.com/hrishin/fabric8-pipeline-library/tree/nodejs

has been taking this approach fyi.

nonetheless we should indeed not run our image manually and instead rely on a dc/rc
